x-pack/filebeat/input/awss3/states.go

```diff
@@ -21,6 +21,7 @@ const (
 )
 
 type listingInfo struct {
+	mu            *sync.Mutex
```

Comment:
`listingInfo` is always pass around as a pointer, so I would have this as a non-pointer field. This gives you an already allocated `sync.Mutex` and gives better locality. It also only needs to protect, the `finalCheck` and `storedObjects` fields, so I would either comment that this is the case or reorders and show the protection by placement.
```
type listingInfo struct {
	totObjects    int
	errorObjects  int

	mu sync.Mutex
	storedObjects int
	finalCheck    bool
}
```